packages/shopify-codemod/transforms/split-return-assignments.js

```diff
@@ -0,0 +1,112 @@
+import {matchLast} from './utils';
+
+export default function splitReturnAssignments({source}, {jscodeshift: j}, {printOptions = {}}) {
+  const sourceAST = j(source);
+  sourceAST
+    .find(j.Function, {
+      body: {
+        body: matchLast(j.ReturnStatement, {
+          type: 'ReturnStatement',
+          argument: {
+            type: 'AssignmentExpression',
+          },
+        }),
+      },
+    })
+    .forEach(({node: {body: {body}}}) => {
+      const {left, right} = body[body.length - 1].argument;
+      const returnLine = body[body.length - 1];
+      if (returnLine.argument.left) {
+        delete body[body.length - 1];
+        body.push({
+          type: 'ExpressionStatement',
+          expression: {
+            type: 'AssignmentExpression',
+            operator: '=',
+            left,
+            right,
+          },
+        });
+
+        if (returnLine.argument.left.name) {
+          body.push({
```

Comment:
You don't need all those cases, you should be able to just have a single case, where you do `body.push(j.returnStatement(returnLine.argument.left))`, which will work regardless of the type of expression that was in the `return` statement.